scripts/sign-commits.sh

```diff
@@ -0,0 +1,35 @@
+#! /usr/bin/env bash
+# SPDX-FileCopyrightText: 2021 TQ Tezos <https://tqtezos.com/>
+#
+# SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+# This script signs all commits in the current 'BUILDKITE_BRANCH'
+
+set -e
+
+git config user.name "serokell-bot" # necessary for pushing
+git config user.email "tezos-packaging@serokell.io"
+git fetch --all
+# We call this script only after a commit was pushed to the given branch, so it's safe to switch to it
+git switch "$BUILDKITE_BRANCH"
+git reset --hard origin/"$BUILDKITE_BRANCH"
+
+# Find if there are unsigned commits in the current branch
+# If there are grep matches then there are unsigned commits
+if git log --pretty="format:%G?" "origin/$BUILDKITE_PIPELINE_DEFAULT_BRANCH..$BUILDKITE_BRANCH" --first-parent | grep "N$"; then
+  echo "Found unsigned commits"
+  # Try to sign and push signed commits, retry in case of collision
+  while : ; do
+    git fetch --all
+    git reset --hard origin/"$BUILDKITE_BRANCH"
```

Comment:
This resets to origin twice from the very start 😞 And also (maybe because it's not quite clear what the rebase does exactly, as I mentioned above) I'm having doubts about resetting after checking that commits need to be signed.
Can there be a following flow?

1. We check `git log` and see some unsigned commits.
2. We reset to origin and there are no more unsigned commits because in that split second another worker has already done that.
3. This script still forges on ahead and tries to sign all the commits.
4. ??? I assume rebase fails and we `exit 1`, but I'm not sure.
5. Even if not, we have performed some unnecessary operations that end up nowhere.

Please correct me if I'm wrong.

Maybe instead, we can inline the `git log` check for unsigned commits into the `while : ; do` loop and re-check after every reset? This would also allow to improve the logging, e.g. we can add `echo "Collision, resetting"` – right now AFAIU it's possible for it to output "Found unsigned commits" and then just fail on rebase. Which, BTW, wouldn't `exit 1` trigger a CI fail? 🤔

Reply:
Granted, this whole scenario is quite unlikely, but I'd like to be sure.

Reply:
> We reset to origin and there are no more unsigned commits because in that split second another worker has already done that.

That's indeed a scenario that I haven't considered. Perhaps, it's worth moving `git log` check inside the loop

Reply:
Updated, please check, it should cover your example now